app/models/goal.py

```diff
@@ -3,4 +3,20 @@
 
 
 class Goal(db.Model):
-    goal_id = db.Column(db.Integer, primary_key=True)
+    goal_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+    # This instance method returns a JSON dictionary of attribute values 
+    # with an optional argument if the goal has tasks assigned to it
+    def convert_to_json(self, tasks_list=None):
```

Comment:
Love this thoughtful, handy optional param here! That said, you shouldn't actually need to ask the caller to pass these in because they are already attached to the goal. See my comment below.